logos-derive/src/graph/regex.rs

```diff
@@ -135,7 +143,7 @@ impl<Leaf: Disambiguate + Debug> Graph<Leaf> {
                     Some(id) => self.merge(id, then),
                     None => then,
                 };
-                let (_, id) = self.parse_mir(*mir, then, Some(miss), None);
+                let (_, id) = self.parse_mir(*mir, then, Some(miss), reserved);
```

Comment:
That's the actual fix to #108, just this one var was not being passed.

It's not being passed in loops either, so a loop that begins with a loop will likely cause the same problem.